python/cuml/internals/global_settings.py

```diff
@@ -23,8 +26,16 @@ class _GlobalSettingsData(threading.local):  # pylint: disable=R0903
 
     def __init__(self):
         super().__init__()
+        if BUILT_WITH_CUDA and has_cuda_gpu():
```

Comment:
Check seems fine, but there will be no way to run `BUILT_WITH_CUDA=True` in a non gpu system, right? 

On the other hand `BUILT_WITH_CUDA=False` in a GPU system is possible, but it might be worth raising a (optionally disable-able) warning that the GPU is not being used. cc @beckernick for opinion on this